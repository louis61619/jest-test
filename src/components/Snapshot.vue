<template>
  <div>
    <div v-if="age < 10">child person</div>
    <div>{{ msg }}</div>
  </div>
</template>
<script>
export default {
  props: {
    msg: {
      type: String,
      default: 'default msg'
    }
  },
  data() {
    return {
      age: 11
    }
  }
}
</script>
