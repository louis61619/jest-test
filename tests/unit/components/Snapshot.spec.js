import { shallowMount } from '@vue/test-utils'
import Snapshot from '@/components/Snapshot.vue'

// npm run test:unit -- -u 進行批量快照更新
// npm run test:unit -- --watch 進行交互式更新

describe('Snapshot.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Snapshot)
  })
  it('match msg snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('match age snapshot', async () => {
    await wrapper.setData({
      age: 6
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
