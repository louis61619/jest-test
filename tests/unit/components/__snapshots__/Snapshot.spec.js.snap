// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot.vue match age snapshot 1`] = `
<div>
  <div>
    child person
  </div>
  <div>
    default msg
  </div>
</div>
`;

exports[`Snapshot.vue match msg snapshot 1`] = `
<div>
  <!--v-if-->
  <div>
    default msg
  </div>
</div>
`;
